pkg/deploymanager/olm.go

```diff
@@ -167,25 +172,23 @@ func (d *DeployManager) DeleteOlmResources(olmResources *OlmResources) error {
 }
 
 // WaitForCatalogSource wait for catalogSource to become ready
-func (d *DeployManager) WaitForCatalogSource() error {
+func (d *DeployManager) WaitForCatalogSource(catalogsource *operatorsv1alpha1.CatalogSource) error {
 
 	timeout := 600 * time.Second
 	interval := 10 * time.Second
 
 	lastReason := ""
 
 	err := utilwait.PollImmediate(interval, timeout, func() (done bool, err error) {
-
-		existingCatalogSource := &operatorsv1alpha1.CatalogSourceList{}
-		err = d.Client.List(d.Ctx, existingCatalogSource, &client.ListOptions{Namespace: InstallNamespace})
+		existingCatalogSource := &operatorsv1alpha1.CatalogSource{}
+		err = d.Client.Get(d.Ctx, client.ObjectKeyFromObject(catalogsource), existingCatalogSource)
 		if err != nil {
-			return false, err
+			lastReason = "failed to get CatalogSource"
```

Comment:
Can you use the fmt.Sprintf of fmt.Errorf and capture the original error as well in the lastReason